rasa/shared/core/domain.py

```diff
@@ -617,6 +626,7 @@ def collect_intent_properties(
         entities: List[Text],
         roles: Dict[Text, List[Text]],
         groups: Dict[Text, List[Text]],
+        default_ignored_entities: List[Text],
```

Comment:
I'm wondering if there's a way we can refactor to avoid adding more parameters to existing functions (in case in the future we want to have more keys in entities or other sections of the domain, functions with over 3 args can become easily unmanageable and unreadable).

One suggestion off the top of my head would be to have a dataclass called `EntityProperties` that contains the 4 parameters as attributes:
```
from dataclasses import dataclass

@dataclass
class EntityProperties:
    """Class for keeping track of the defined entities in the domain."""
    entities: List[Text]
    roles: Dict[Text, List[Text]]
    groups: Dict[Text, List[Text]]
    default_ignored_entities: List[Text]
```
Then `collect_entity_properties` would return an instance of this data class which then gets used as parameter everywhere else?

Reply:
Makes sense, I'll try it out and see what the impact will be elsewhere

Reply:
Alrighty, works well enough! I notice that `intent_properties` is similar - considered refactoring it in the same way, but it seems overkill since the only preserved property is `used_entities` in the end.